sycl/doc/extensions/proposed/sycl_ext_oneapi_prefetch.asciidoc

```diff
@@ -0,0 +1,655 @@
+    foo(data[j]);
+  }
+});
+----
+
+=== Group prefetches
+
+The functions in this section allow groups of work-items to cooperatively
+prefetch the same data. These functions are all group functions, as defined
+in Section 4.17.3 of the SYCL specification.
+
+[NOTE]
+====
+Although calling `joint_prefetch` is functionally equivalent to calling
+`prefetch` from every work-item in a group, some implementations may be able
+to issue cooperative prefetches more efficiently on some hardware.
+====
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename Group, typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, void* ptr, Properties properties = {});
+
+template <typename Group, typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, void* ptr, size_t bytes, Properties properties = {});
+
+template <typename Group, typename T, typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, T* ptr, Properties properties = {});
+
+template <typename Group, typename T, typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, T* ptr, size_t count, Properties properties = {});
+
+// Only available if AddressSpace == global_space || AddressSpace == generic_space
+template <typename Group, access::address_space AddressSpace, access::decorated IsDecorated,
+          typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, multi_ptr<void, AddressSpace, IsDecorated> ptr,
+                    Properties properties = {});
+
+// Only available if AddressSpace == global_space || AddressSpace == generic_space
+template <typename Group, access::address_space AddressSpace, access::decorated IsDecorated,
+          typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, multi_ptr<void, AddressSpace, IsDecorated> ptr, size_t bytes,
+                    Properties properties = {});
+
+// Only available if AddressSpace == global_space || AddressSpace == generic_space
+template <typename Group, typename T,
+          access::address_space AddressSpace, access::decorated IsDecorated,
+          typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, multi_ptr<T, AddressSpace, IsDecorated> ptr,
+                    Properties properties = {});
+
+// Only available if AddressSpace == global_space || AddressSpace == generic_space
+template <typename Group, typename T,
+          access::address_space AddressSpace, access::decorated IsDecorated,
+          typename Properties = ext::oneapi::experimental::properties<>>
+void joint_prefetch(Group g, multi_ptr<T, AddressSpace, IsDecorated> ptr, size_t count,
+                    Properties properties = {});
+
+// Only available if Dimensions == 1 && (AccessMode == read || AccessMode == read_write)
```

Comment:
```suggestion
// Only available if Dimensions > 0 && (AccessMode == read || AccessMode == read_write)
```

And the next one below too.

Reply:
Good catch!  Fixed in 9e43051f8482878b2d04b662d95a03e9670bbdee.